paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -417,29 +412,38 @@ internal class PaymentSheetViewModelTest {
             val selection = PaymentSelection.Saved(PaymentMethodFixtures.CARD_PAYMENT_METHOD)
             viewModel.updateSelection(selection)
 
-            val viewState: MutableList<PaymentSheetViewState?> = mutableListOf()
-            viewModel.viewState.observeForever {
-                viewState.add(it)
-            }
+            val resultTurbine = viewModel.paymentSheetResult.testIn(this)
+            val viewStateTurbine = viewModel.viewState.testIn(this)
 
-            viewModel.paymentSheetResult.test {
-                viewModel.onPaymentResult(PaymentResult.Completed)
-                assertThat(viewState[1])
-                    .isInstanceOf(PaymentSheetViewState.FinishProcessing::class.java)
-                (viewState[1] as PaymentSheetViewState.FinishProcessing).onComplete()
-                assertThat(awaitItem())
-                    .isEqualTo(PaymentSheetResult.Completed)
-            }
+            viewModel.onPaymentResult(PaymentResult.Completed)
+
+            assertThat(viewStateTurbine.awaitItem())
+                .isEqualTo(PaymentSheetViewState.Reset(null))
+
+            val finishedProcessingState = viewStateTurbine.awaitItem()
+            assertThat(finishedProcessingState)
+                .isInstanceOf(PaymentSheetViewState.FinishProcessing::class.java)
+
+            (finishedProcessingState as PaymentSheetViewState.FinishProcessing).onComplete()
+
+            assertThat(resultTurbine.awaitItem())
+                .isEqualTo(PaymentSheetResult.Completed)
 
             verify(eventReporter)
                 .onPaymentSuccess(selection)
-
             assertThat(prefsRepository.paymentSelectionArgs)
                 .containsExactly(selection)
-            assertThat(prefsRepository.getSavedSelection(true, true))
-                .isEqualTo(
-                    SavedSelection.PaymentMethod(selection.paymentMethod.id.orEmpty())
+            assertThat(
+                prefsRepository.getSavedSelection(
+                    isGooglePayAvailable = true,
+                    isLinkAvailable = true
                 )
+            ).isEqualTo(
+                SavedSelection.PaymentMethod(selection.paymentMethod.id.orEmpty())
+            )
+
+            resultTurbine.cancel()
```

Comment:
I'm curious, why do you cancel the turbines?

Reply:
https://github.com/cashapp/turbine#multiple-flows, need to complete, otherwise test will hang